pkg/controller/nodes/dynamic/dynamic_workflow.go

```diff
@@ -171,27 +201,53 @@ func (d dynamicNodeTaskNodeHandler) buildContextualDynamicWorkflow(ctx context.C
 		return dynamicWorkflowContext{}, errors.Wrapf(utils.ErrorCodeSystem, err, "unable to read futures file, maybe corrupted")
 	}
 
+	closure, dynamicWf, workflowContext, err := d.buildDynamicWorkflow(ctx, nCtx, djSpec, dynamicNodeStatus)
+	if err != nil {
+		return workflowContext, err
+	}
+
+	if err := f.Cache(ctx, dynamicWf, closure); err != nil {
+		logger.Errorf(ctx, "Failed to cache Dynamic workflow [%s]", err.Error())
+	}
+
+	// The current node would end up becoming the parent for the dynamic task nodes.
```

Comment:
do you need to call setEphemeralNodeExecutionStatusAttributes here too?

Reply:
in the case of the cache, sadly, GetNodeExecutionStatus has a side-effect that sets those attributes